### ISSUE TITLE
Bitmessage hangs when 'disk full' condition occurrs

### ISSUE THREAD

**Issue by yurivict** [Bitmessage/PyBitmessage#572]
I accidentally had the 'disk full' condition on the disk where bitmessage stored data. After deleting large files and clearing this condition, bitmessage remained frozed with 0% CPU use. GUI was repainting, but not clickable.

All other running programs continued fine, but bitmessage.

**Comment by PeterSurda** [comment]
I'll see if I can reproduce and fix this.

**Comment by yurivict** [comment]
Now I got disk full with 0.6.0
Problems:
* It stops with not one but two pop messages. There is no need to have more than one
* Can't close the popup, pressing the button doesn't do anything

Also, it doesn't need to stop permanently, because disk full condition can later go away, The correct behavior is to check the available disk space every minute, and resume operation if there is significant space available.

**Comment by bmng-dev** [comment]
A disk full condition is commonly encountered when VACUUMing the database because according to the [VACUUM documentation](https://sqlite.org/lang_vacuum.html):
> as much as twice the size of the original database file is required in free disk space

I think going forward VACUUM should be performed sparingly and only when the user allows it. This would require a non-modal notification/prompt. In place of the current usage of VACUUM, [auto_vacuum](https://sqlite.org/pragma.html#pragma_auto_vacuum) could be set to incremental and [incremental_vacuum](https://sqlite.org/pragma.html#pragma_incremental_vacuum) performed (after ensuring [freelist_count](https://sqlite.org/pragma.html#pragma_freelist_count) is greater than zero) whenever the user invokes 'Delete all trashed messages' and the cleaner thread purges the inventory of expired items. Unfortunately enabling auto_vacuum for existing users will require the use of VACUUM

All this does not resolve the issue of how PyBitmessage behaves when the disk is full but it should alleviate the occurance in the short term so a proper solution can be implemented.

**Comment by PeterSurda** [comment]
I'll look at this, but other than fixing the popup I probably won't do anything else (e.g. fiddling around with VACUUM). Have your computer notify you when you run low on disk space.

**Comment by yurivict** [comment]
The user doesn't look at notifications all the time. I come back in a few hours, plenty of disk space is available, but BM displays the messages that can't be closed and needs to be killed.

**Comment by PeterSurda** [comment]
Looking at the source, PyBitmessage is supposed to shutdown if disk is full, but I think it freezes instead because only the SQL thread ends while other keep running. A clean shutdown cannot happen because it needs to writes data.

If instead the SQL thread would wait, then the problem with inability to cleanly shutdown remains. The disk being full can also cause problems with updating keys.dat and known addresses when triggered by other means than during the shutdown. It also cannot log anything and even communicate correctly with other nodes because it cannot update its own inventory.

So to sum it up, I think that it should immediately quit and not even try to display the message, just like it does when it runs without the GUI.

**Comment by PeterSurda** [comment]
@yurivict use other sysadmin tools to prevent disk being full, if notifications don't work then you can use quotas.
